dev-guide.adoc

```diff
@@ -6,38 +6,22 @@ The bulk of the logic is in package `utils`. Here, we have three generic interfa
 
 * `operations.go` has abstracted functions, and is how implementations of the puller should interact with the cluster. These functions handle resource creation/deletion and also ensuring no issues occur in that process.
 * `clusterutils.go` is a lower-level interface to the cluster, meant to be called from `operations.go` functions. It is responsible for holding k8s-specific logic, actually interfacing with the cluster API, and waiting for resources to be created.
-* `auth.go` contains logic for authenticating with fabric8-auth and getting tokens for users, and is only used in the multi-cluster case.
 
-The functions in `operations.go` are called from `puller.go` files in `pkg`, with separate cases for single and multi cluster.
+The functions in `operations.go` are called from `puller.go` files in `pkg`.
 
-Implementations in `pkg` basically create a k8s client config, get auth data if necessary), set up some channels, and start goroutines that recreates the daemonset on a delay. In the multicluster case, we start a goroutine for each cluster we're caching images on.
-
-At a high level, the flow in multi-cluster mode is
-
-. Get the rh-che service account token using the secret used for rh-che. Use this service account token for authenticating with fabric-oso-proxy
-. For each user in `cfg.ImpersonateUsers`, start a goroutine to create a daemonset impersonating that user.
-** The users to impersonate are chosen such that each user proxies to a different tenant cluster
-** The fabric8-oso-proxy will use the rhche bearer token to get the correct token for manipulating the tenant clusters, and the impersonate header to proxy the request to the correct cluster.
-
-== Single cluster vs Multi cluster
-The image puller is designed to run on the che.openshift.io clusters; as such it includes a fair bit of logic for running in the production cluster and creating daemonsets on tenant clusters.
-
-The key logic for creating daemonsets is split between `pkg/multi-cluster`, which handles the production case, and `pkg/single-cluster`, which can run anywhere to test basic things.
-
-This means that testing changes can be tricky -- you can't run the multi-cluster code locally, and the single-cluster configuration won't necessarily work perfectly in the multi-cluster case.
+Implementations in `pkg` basically create a k8s client config, set up some channels, and start goroutines that recreates the daemonset on a delay.
 
 == Code structure
-* `cfg` - read configuration from env vars into global vars. This can be accessed anywhere via e.g. `cfg.MultiCluster`
 
-* `cmd` - Entrypoint for service; switches between single cluster and multi cluster mode.
+* `cfg` - read configuration from env vars into global vars. This can be accessed by running `cfg.GetConfig()`
```

Comment:
"running" here is kind of strange, since this is in the context of writing code. Perhaps "calling" is better?

Also, `cfg` no longer uses global variables, and instead encapulates config into a `Config` struct.